frontend/src/common/queries/collections.ts

```diff
@@ -65,36 +80,93 @@ async function fetchCollection(
   _: unknown,
   id: string,
   visibility: VISIBILITY_TYPE
-): Promise<Collection | null> {
+): Promise<Map<VISIBILITY_TYPE, Collection> | null> {
   if (!id) {
     return null;
   }
 
+  const collections = new Map<VISIBILITY_TYPE, Collection>();
+
   const baseUrl = apiTemplateToUrl(API_URL + API.COLLECTION, { id });
 
   const finalUrl =
     visibility === VISIBILITY_TYPE.PRIVATE
       ? baseUrl + `?visibility=${VISIBILITY_TYPE.PRIVATE}`
       : baseUrl;
 
-  const response = await fetch(finalUrl, DEFAULT_FETCH_OPTIONS);
-  const result = await response.json();
+  let response = await fetch(finalUrl, DEFAULT_FETCH_OPTIONS);
+  let json = await response.json();
 
   if (!response.ok) {
-    throw result;
+    throw json;
   }
 
-  return result;
+  let datasetMap = new Map() as Collection["datasets"];
+  for (const dataset of json.datasets) {
+    datasetMap.set(dataset.original_id || dataset.id, dataset);
+  }
+
+  collections.set(visibility, { ...json, datasets: datasetMap });
```

Comment:
Store datasets as`Map<Dataset["id"], Dataset>` instead of `Array` for easy comparison between published collection and private revision